### DISCUSSION TITLE
TimeInput's onChange value should be a string instead of a ChangeEvent<HTMLInputElement>?

### DISCUSSION THREAD

**Discussion by julamb** [mantinedev/mantine#4395] [Feedback]
That would be more consistent with `DateInput` returning a Date, `NumberInput` returning a number and `TextInput` returning a string, and would make it easier to use with `useForm` & co, what do you think?

**Comment by rtivital** [comment]
1. TextInput does not call onChange with `string`, it is called with event
2. TimeInput is a thin wrapper of `input[type="time"]` – you may need this event object in some cases, for example to prevent default

**Comment by julamb** [comment]
Makes sense!
